packages/http/README.md

```diff
@@ -1,3 +1,405 @@
+  validate?: {
+    request?:
+      | boolean
+      | {
+          hijack?: boolean;
+          headers?: boolean;
+          query?: boolean;
+          body?: boolean;
+        };
+
+    response?:
+      | boolean
+      | {
+          headers?: boolean;
+          body?: boolean;
+        };
+  };
+}
+```
+
+#### Config Examples
+
+**Do not mock and do not validate requests**
+
+```javascript
+const config = {
+  mock: false,
+  validate: {
+    request: false,
+    // response is set to true by default
+  },
+};
+```
+
+With the above configuration the http client will proxy your requests to a server and will validate the response with the OpenAPI you loaded. However, it will not validate the input (e.g. will not check whether the provided query param is valid).
+
+**Validate only request's body and response's headers**
+
+```javascript
+const config = {
+  validate: {
+    request: {
+      body: true,
+    },
+    response: {
+      headers: true,
+    },
+  },
+};
+```
+
+**When mocking a response generate static response**
+
+```javascript
+const config = {
+  mock: {
+    dynamic: false,
+  },
+};
+```
+
+Generating static responses means that if an OpenAPI operation's response has any examples defined the payload we construct will use those examples.
+
+This contrasts "dynamic responses" which means generating responses from a json schema.
+
+**Return named example of a specified key**
+
+```javascript
+const config = {
+  mock: {
+    example: 'key',
+  },
+};
+```
+
+In OpenAPI Specification 3 one can construct such response that will have multiple examples mapped by keys.
+This configuration allows you to be very specific which example you want to choose.
+
+**Return 403 response of specified mime type**
+
+```javascript
+const config = {
+  mock: {
+    code: 403,
+    mimeType: 'application/xml',
+  },
+};
+```
+
+This will enforce a 403 response (given that such response is defined in your OpenAPI file).
+
+## Loading specs
+
+When you load a spec you "teach" prism how your API looks like.
+You can run prism without loading a spec, but features such as validation will be disabled.
+
+```javascript
+const loaderConfig = { path: '/absolute/file/path.oas3.yaml' };
+const promise = prism.load(loaderConfig);
+```
+
+## Making requests
+
+To make a basic request you need to do the following
+
+```javascript
+const request = {
+  method: 'get',
+  url: {
+    path: '/path', // must be prefixed with slash
+  },
+};
+const promise = prism.process(request);
+```
+
+The request object has the following interface
+
+```ts
+export interface IHttpRequest {
+  method: IHttpMethod;
+  url: IHttpUrl;
+  headers?: IHttpNameValue;
+  body?: any;
+}
+```
+
+All of this is pretty standard except the `url.baseUrl` which we will describe in more details.
+
+### Server Validation
+
+Consider a request
+
+```javascript
+prism.process({
+  method: 'get',
+  url: {
+    path: '/path', // must be prefixed with slash
+    baseUrl: 'http://localhost:4010',
```

Comment:
Are we keeping localhost:4010 here for a reason? We changed the other instances to a real server for clarity, is this different ?

Reply:
we are not, I'm dumb :P